### ISSUE TITLE
[FEA] Support Cache(PCBS) Array-of-Struct

### ISSUE THREAD

**Issue by viadea** [NVIDIA/spark-rapids#3573]
**Is your feature request related to a problem? Please describe.**
I wish the plugin can support Cache(PCBS) Array-of-Struct.

**Describe the solution you'd like**
Currently if we enable PCBS to cache an Array-Of-Struct, it will fallback on CPU.

Minimum repro:
```
import org.apache.spark.sql.Row
import org.apache.spark.sql.types._

val data = Seq(
    Row(Row("Adam ","","Green"),1,"M",10),
    Row(Row("Bob ","Middle","Green"),2,"M",20),
    Row(Row("Cathy ","","Green"),3,"F",30)
)

val schema = (new StructType()
  .add("name",new StructType()
    .add("firstname",StringType)
    .add("middlename",StringType)
    .add("lastname",StringType)) 
  .add("low",IntegerType)
  .add("gender",StringType)
  .add("high",IntegerType))

val df = spark.createDataFrame(spark.sparkContext.parallelize(data),schema)
df.write.format("parquet").mode("overwrite").save("/tmp/testparquet")

val df2 = spark.read.parquet("/tmp/testparquet")
df2.createOrReplaceTempView("df2")

val array_of_struct=spark.sql("SELECT array(struct(low,high)) as arrayofstruct FROM df2")
array_of_struct.write.format("parquet").mode("overwrite").save("/tmp/testparquet2")
val array_of_struct2 = spark.read.parquet("/tmp/testparquet2")
array_of_struct2.cache()
array_of_struct2.collect()
```

Driver log:
```bash
!Exec <InMemoryTableScanExec> cannot run on GPU because unsupported data types in output: ArrayType(StructType(StructField(low,IntegerType,true), StructField(high,IntegerType,true)),true) [arrayofstruct#85]
```

**Describe alternatives you've considered**

**Additional context**
Add any other context, code examples, or references to existing implementations about the feature request here.

**Comment by razajafri** [comment]
depends on #2054

**Comment by razajafri** [comment]
closed as part of #3619
